pkg/doc/did/doc.go

```diff
@@ -1004,3 +1295,24 @@ func BuildDoc(opts ...DocOption) *Doc {
 
 	return doc
 }
+
+func createDocumentLoader() ld.DocumentLoader {
+	loader := ld.NewCachingDocumentLoader(ld.NewRFC7324CachingDocumentLoader(&http.Client{}))
```

Comment:
Hardcoding the http client makes this harder to test and may impact future work that requires different implementations. I've already been bit by this sort of issue while working on unit tests and examples.

Reply:
Can't there be an option where you can provide your own document loader?

Reply:
Sure, I will add an option for this, at https://github.com/hyperledger/aries-framework-go/issues/1846.